src/environment/mod.rs

```diff
@@ -104,6 +105,18 @@ pub trait Environment: 'static + Clone + Debug + Default + Send + Sync {
         static TERMINATOR: OnceCell<Arc<AtomicBool>> = OnceCell::new();
         TERMINATOR.get_or_init(|| Arc::new(AtomicBool::new(false)))
     }
+
+    /// Returns a thread pool for the node to perform intensive operations.
+    fn thread_pool() -> &'static Arc<ThreadPool> {
+        static POOL: OnceCell<Arc<ThreadPool>> = OnceCell::new();
+        POOL.get_or_init(|| {
+            Arc::new(ThreadPoolBuilder::new()
+                .stack_size(8 * 1024 * 1024)
+                .num_threads((num_cpus::get() * 7 / 8).max(2))
```

Comment:
Note that this will result in a lot of threads dedicated to networking on systems with a large number of cores.

Reply:
This threadpool is only applicable to mining-related operations; the network bits are handled by the `tokio` threadpool  (minus some (de)serialization that is handled via a global `rayon` threadpool); it's easy to get them mixed up, though, and this PR simplifies things a bit; it might be a good idea to adjust this threadpool's name so that it's more clear that it's a dedicated `rayon` threadpool (as opposed to the `tokio` or global `rayon` one).